src/core/core.layouts.js

```diff
@@ -220,189 +378,80 @@ module.exports = {
 		// 9. Tell any axes that overlay the chart area the positions of the chart area
 
 		// Step 1
-		var chartWidth = width - leftPadding - rightPadding;
-		var chartHeight = height - topPadding - bottomPadding;
-		var chartAreaWidth = chartWidth / 2; // min 50%
-
-		// Step 2
-		var verticalBoxWidth = (width - chartAreaWidth) / verticalBoxes.length;
-
-		// Step 3
-		// TODO re-limit horizontal axis height (this limit has affected only padding calculation since PR 1837)
-		// var horizontalBoxHeight = (height - chartAreaHeight) / horizontalBoxes.length;
+		var chartHeight = height - padding.height;
+		var chartArea = {
+			w: width - padding.width,
+			h: chartHeight
+		};
+		var chartAreaWidth = chartArea.w / 2; // min 50%
+
+		// Steps 2-3
+		var params = {
+			outerWidth: width,
+			outerHeight: height,
+			padding: padding,
```

Comment:
Can we call this `userPadding` or `paddingOpts`?

Reply:
Done `userPadding`